.github/workflows/linter.yml

```diff
@@ -1,27 +1,50 @@
 name: Linter
 
 on:
-    push:
-        branches:
-            - main
-    pull_request:
-        branches:
-            - "*"
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "*"
 
 jobs:
-    solhint:
-        runs-on: ubuntu-20.04
-        steps:
-            - name: Checkout teleporter
-              uses: actions/checkout@v4
+  solhint:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout teleporter
+      uses: actions/checkout@v4
 
-            - name: Install solhint
-              run: |
-                  npm install solhint -g
-                  solhint --version
+    - name: Install solhint
+      run: |
+        npm install solhint -g
+        solhint --version
 
-            # "solhint **/*.sol" runs differently than "solhint '**/*.sol'", where the latter checks sol files
-            # in subdirectories. The former only checks sol files in the current directory and directories one level down.
-            - name: Run Lint
-              run: |
-                  ./scripts/lint.sh
+    # "solhint **/*.sol" runs differently than "solhint '**/*.sol'", where the latter checks sol files
+    # in subdirectories. The former only checks sol files in the current directory and directories one level down.
+    - name: Run Lint
+      run: |
+        ./scripts/lint.sh
+
+  golangci:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        path: teleporter
+        submodules: recursive
+
+    - name: Setup Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.20.10'
+        check-latest: true
+
+    - name: golangci-lint
```

Comment:
should we add golangci-lint for `lint.sh` script

Reply:
Good call. Added.

Reply:
was this added? Was thinking `/scripts/lint.sh` would also run golangci-lint

Reply:
I did add it to `lint.sh`, but then realized that the solhint job would fail since it does not install golangci-lint. It didn't make much sense to install golangci-lint in the solidity lint job, so I reverted that change.

It makes logical sense to have separate Go and Solidity lint jobs. It also makes sense to have a single unified lint.sh scripts that covers both languages. If we can make it clear what linter is failing, my vote would be to unify the lint.sh script and the lint job.

Reply:
maybe we can do this with command line flags to `lint.sh`